lib/search/grouped_search_results.rb

```diff
@@ -133,5 +146,11 @@ def self.blurb_for(cooked: nil, term: nil, blurb_length: BLURB_LENGTH, scrub: tr
       blurb = TextHelper.truncate(cooked, length: blurb_length) if blurb.blank?
       Sanitize.clean(blurb)
     end
+
+    private
+
+    def add_topic_category(category)
+      topic_categories << category if category && !topic_categories.include?(category)
```

Comment:
> topic_categories.include?(category)

Maybe just use a set here instead which is more efficient that having to compare against all the objects in the existing array. 

> topic_categories << category

I also think we can just include the parent category here so that we don't have to make two method calls. 

```
topic_categories << category
topic_categories << category.parent_category if category&.parent_category
```

Reply:
The order is important here, but that's not a problem because `Set`s are ordered in Ruby.